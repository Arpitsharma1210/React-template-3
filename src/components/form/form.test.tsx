import React from 'react';
import renderer from 'react-test-renderer';
import { Form, FormRow, FormRowItem, FormError } from './index';

describe('Form Component', () => {
  it('renders children correctly', () => {
    const tree = renderer
      .create(
        <Form>
          <div>Child Element</div>
        </Form>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders FormRow component correctly', () => {
    const tree = renderer
      .create(
        <FormRow>
          <div>Child Element</div>
        </FormRow>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders FormRowItem component correctly', () => {
    const tree = renderer
      .create(
        <FormRowItem>
          <div>Child Element</div>
        </FormRowItem>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders FormError component correctly with custom message', () => {
    const tree = renderer
      .create(<FormError message="Custom Error Message" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('FormRow Component', () => {
  it('renders children correctly', () => {
    const tree = renderer
      .create(
        <FormRow>
          <div>Test Child</div>
        </FormRow>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('FormRowItem Component', () => {
  it('renders children correctly', () => {
    const tree = renderer
      .create(
        <FormRowItem>
          <div>Test Child</div>
        </FormRowItem>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Department constant', () => {
  it('contains correct label and value pairs', () => {
    const department = [
      { label: "HR", value: "hr" },
      { label: "Design", value: "design" },
      { label: "Engineering", value: "engineering" },
      { label: "Business", value: "business" },
    ];

    department.forEach(({ label, value }) => {
      expect(label).toBeDefined();
      expect(value).toBeDefined();
    });
  });
});
