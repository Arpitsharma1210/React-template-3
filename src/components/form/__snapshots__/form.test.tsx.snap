// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component renders FormError component correctly with custom message 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard sc-hLljAB kDJCrq css-1hdyuqk-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ErrorOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    Custom Error Message
  </div>
</div>
`;

exports[`Form Component renders FormRow component correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container sc-gsMHZj kyEOGR css-11lq3yg-MuiGrid-root"
>
  <div>
    Child Element
  </div>
</div>
`;

exports[`Form Component renders FormRowItem component correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item sc-dkHyXG cUUUEY css-13i4rnv-MuiGrid-root"
>
  <div>
    Child Element
  </div>
</div>
`;

exports[`Form Component renders children correctly 1`] = `
<form
  className="sc-bcHOTY jCvqer"
  noValidate={true}
>
  <div>
    Child Element
  </div>
</form>
`;

exports[`FormRow Component renders children correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container sc-gsMHZj kyEOGR css-11lq3yg-MuiGrid-root"
>
  <div>
    Test Child
  </div>
</div>
`;

exports[`FormRowItem Component renders children correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item sc-dkHyXG cUUUEY css-13i4rnv-MuiGrid-root"
>
  <div>
    Test Child
  </div>
</div>
`;
